### PR TITLE
Omit `webdrivers` gem from `Gemfile` template

### PR DESCRIPTION
**Pull request by seanpdoyle**
### Motivation / Background

As of Selenium 4.6, [the Selenium Manager is capable of managing Chrome Driver installations and integrations][readme]. As of Selenium 4.11, the
Selenium Manager is capable of [capable of resolving the Chrome for
Testing installation][] path.

### Detail

By omitting the `gem` declaration from the `Gemfile.tt`, newly generated applications and applications updating their `Gemfile` in lockstep with newer Rails versions can shed the dependency and avoid test failures introduced by newly released Chrome versions (like, for example, [titusfortner/webdrivers#247][]).

[readme]: https://github.com/titusfortner/webdrivers/tree/43f8ac436cc4121c903c1c611dfe76088ef6cbab#update-selenium-manager
[titusfortner/webdrivers#247]: https://github.com/titusfortner/webdrivers/issues/247
[capable of resolving the Chrome for Testing installation]: https://github.com/rails/rails/pull/48847#issuecomment-1656756862

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] Tests are added or updated if you fix a bug or add a feature.
* [x] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

### CODE REVIEW

**Comment by simi** [comment]
Wouild it make sense to ensure selenium-webdriver 4.6+ gem is installed in [Gemfile](https://github.com/rails/rails/blob/be3f4e6e1ba954ce7c36913939eb802dc23292e2/Gemfile#L18C6-L18C24)?

**Comment by seanpdoyle** [comment]
> Wouild it make sense to ensure selenium-webdriver 4.6+ gem is installed in [Gemfile](https://github.com/rails/rails/blob/be3f4e6e1ba954ce7c36913939eb802dc23292e2/Gemfile#L18C6-L18C24)?

That's a great idea. I've changed both Rails' development `Gemfile` (the one you've linked to here) as well as the `Gemfile.tt` template file to specify `>= 4.6.0`.

**Comment by titusfortner** [comment]
It should be Selenium 4.11 (not yet released) because while 4.6 - 4.10 manages drivers, it does not support the new Chrome for Testing location for chromedriver 115+.

**Comment by seanpdoyle** [comment]
@titusfortner thank you for providing that context!

I'd like to note that CI was passing when locking to version 4.6, but will fail indefinitely until [Selenium releases 4.11](https://github.com/SeleniumHQ/selenium/releases).

I'm now subscribed for those releases, so I'll come back to ping this PR once that occurs (@titusfortner I'll be you're also patiently watching and waiting, so don't hesitate to ping this PR if you beat me to it).

**Comment by guilleiguaran** [comment]
Looks like we need to use a lower version for Ruby 2.7:

```
Could not find compatible versions
--
  |  
  | Because selenium-webdriver >= 4.9.1 depends on Ruby >= 3.0
  | and Gemfile depends on selenium-webdriver >= 4.11.0,
  | Ruby >= 3.0 is required.
  | So, because current Ruby version is = 2.7.8,
  | version solving has failed.
```

**Comment by seanpdoyle** [comment]
@guilleiguaran thank you for sharing that, that's good information!

It's probably a discussion worth having outside of this branch, but [Ruby 2.7 is end-of-life](https://www.ruby-lang.org/en/downloads/branches/) as of 2023-03-31.

File: Gemfile [hunk 1]
```diff
@@ -15,7 +15,7 @@ gem "rake", ">= 13"
 gem "sprockets-rails", ">= 2.0.0"
 gem "propshaft", ">= 0.1.7"
 gem "capybara", ">= 3.38"
```
Comment (titusfortner) [inline Gemfile:17]: Selenium 4.10+ needs capybara 3.39+

File: actionpack/lib/action_dispatch/system_testing/browser.rb [hunk 2]
```diff
@@ -26,7 +26,7 @@ def configure
         yield options if block_given? && options
       end
 
-      # driver_path can be configured as a proc. The webdrivers gem uses this
+      # driver_path can be configured as a proc. The selenium-webdriver gem uses this
```
Comment (titusfortner) [inline actionpack/lib/action_dispatch/system_testing/browser.rb:29]:
```suggestion
      # driver_path can be configured as a proc.
```

Comment (titusfortner) [reply actionpack/lib/action_dispatch/system_testing/browser.rb:29]: (Selenium is what executes the proc)

**Comment by titusfortner** [comment]
Selenium supports what Ruby supports, and I'll strongly defend that as a feature and not a bug for any and all Ruby open source projects.

Alternately, webdrivers 5.3 now has support for chromedriver 115+ and still works for Ruby 2.7
The downside is that it restricts Selenium 4.11+ and gives a post_install_message
That might be better at this point, though?

**Comment by rafaelfranca** [comment]
We can't drop support to old rubies without bumping the major version of Rails. We are not going to bump the major version of Rails for the next release so we need to support Ruby 2.7.

File: railties/lib/rails/generators/rails/app/templates/Gemfile.tt [hunk 3]
```diff
@@ -59,8 +59,7 @@ end
 <%- if depends_on_system_test? -%>
 group :test do
   # Use system testing [https://guides.rubyonrails.org/testing.html#system-testing]
-  gem "capybara"
-  gem "selenium-webdriver"
-  gem "webdrivers"
+  gem "capybara", ">= 3.39"
+  gem "selenium-webdriver", ">= 4.11.0"
```
Comment (rafaelfranca) [inline railties/lib/rails/generators/rails/app/templates/Gemfile.tt:63]:
```suggestion
  gem "capybara"
  gem "selenium-webdriver"
```

New applications don't have a Gemfile.lock yet, so they will always get the most up-to-date gem.

Comment (rafaelfranca) [reply railties/lib/rails/generators/rails/app/templates/Gemfile.tt:63]: If `webdrivers` is necessary in the version that support Ruby 2.7 we will need to conditionally add that gem when the app is generated on that Ruby version.

**Pull request merged**